contracts/StandardBounty.sol

```diff
@@ -0,0 +1,319 @@
+pragma solidity ^0.4.18;
+import "./inherited/StandardToken.sol";
+
+/// @title StandardBounties
+/// @dev Used to pay out individuals or groups for task fulfillment
+
+contract StandardBounty {
+
+  /*
+   * Storage
+   */
+
+  address public masterCopy;
+
+  address public controller;
+  string public data;
+
+  bool public hasPaidOut;
+
+  Fulfillment[] public fulfillments;
+
+  Contribution[] public contributions;
+
+
+
+  /*
+   * Events
+   */
+  event BountyFulfilled(uint256 _fulfillmentId);
+  event FulfillmentUpdated(uint _fulfillmentId);
+  event FulfillmentAccepted(uint256 _fulfillmentId);
+  event BountyDrained(address controller);
+  event BountyChanged();
+
+  /*
+   * Structs
+   */
+  struct Fulfillment {
+      address[] fulfillers;
+      uint[] numerators;
+      uint denomenator;
+      string data;
+      bool accepted;
+  }
+
+  struct Contribution {
+      address contributor;
+      uint[] amounts;
+      StandardToken[] tokens;
+      bool refunded;
+      mapping (address => bool) hasContributed;
+  }
+
+  /*
+   * Modifiers
+   */
+
+  modifier onlyController() {
+      require(msg.sender == controller);
+      _;
+  }
+
+  modifier validateFulfillmentArrayIndex(uint _index) {
+      require(_index < fulfillments.length);
+      _;
+  }
+
+  modifier validateNotTooManyFulfillments(){
+    require((fulfillments.length + 1) > fulfillments.length);
```

Comment:
hmm this value is pretty high. so this would be intentional attacks. interesting.

Reply:
Do we not want to set a fixed size to be safe?